runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -171,6 +193,83 @@ public static Environment createProcessEnvironment(
     }
   }
 
+  public static Collection<ArtifactInformation> getArtifacts(PipelineOptions options) {
```

Comment:
Is this method just to get artifacts from the experiment ? If so please name this method accordingly.

Reply:
It gets a list of all artifacts from the experiment as well as from ClassLoader. The method was moved from `PortableRunner` to pre-generate the dependencies (originally the dependencies were calculated when submitting pipeline).

Reply:
It gets a list of all artifacts from the experiment as well as from ClassLoader. The method was moved from `PortableRunner` to pre-generate the dependencies (originally the dependencies were calculated when submitting pipeline).